src/psyclone/domain/nemo/transformations/create_nemo_psy_trans.py

```diff
@@ -0,0 +1,173 @@
+                            BinaryOperation[operator:'MUL']
+                                Literal[value:'2.0', Scalar<REAL, UNDEFINED>]
+                                Reference[name:'ji']
+
+    The result of this transformation is that the root `Routine` has
+    been converted into a `NemoInvokeSchedule`, the `Loop` is now a
+    `NemoLoop` (with type 'lon' [for longitude]) and the body of the loop
+    is now an `InlinedKern`.
+
+    """
+    @property
+    def name(self):
+        '''
+        :returns: the name of the transformation.
+        :rtype: str
+
+        '''
+        return type(self).__name__
+
+    def validate(self, node, options=None):
+        '''
+        Check that the supplied node is a valid target for this transformation.
+
+        :param node: the root of the PSyIR tree to be transformed.
+        :type node: :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for \
+            transformations. No options are used in this \
+            transformation. This is an optional argument that defaults \
+            to None.
+        :type options: dict of string:values or None
+
+        :raises TransformationError: if the supplied node is not a PSyIR node.
+
+        '''
+        if not isinstance(node, Node):
+            raise TransformationError(
+                "Error in CreateNemoPSyTrans transformation. The supplied node"
+                " should be a PSyIR Node but found '{0}'".format(
+                    type(node).__name__))
+
+        super(CreateNemoPSyTrans, self).validate(node, options=options)
+
+    def apply(self, psyir, options=None):
+        '''
+        Takes generic PSyIR and replaces recognised structures with
+        NEMO-specific PSyIR (in-place). Note that this may mean replacing
+        the top-level node itself and therefore this routine returns the
+        root of the modified tree.
+
+        :param psyir: the root node of the PSyIR tree to process.
+        :type psyir: :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for \
+            transformations. No options are used in this \
+            transformation. This is an optional argument that defaults \
+            to None.
+        :type options: dict of string:values or None
+
+        :returns: root of the modified PSyIR tree.
+        :rtype: :py:class:`psyclone.psyir.nodes.Node`
+
+        '''
+        self.validate(psyir, options=options)
+
+        invoke_trans = CreateNemoInvokeScheduleTrans()
+        kern_trans = CreateNemoKernelTrans()
+        loop_trans = CreateNemoLoopTrans()
+
+        # Deal with Routines first. Have to take care of the case where the
+        # supplied top-level node is itself a Routine and must therefore be
+        # replaced.
+        root = psyir
+
+        for routine in psyir.walk(Routine):
+            new_node = invoke_trans.apply(routine)
+            if routine is root:
+                root = new_node
```

Comment:
I think with the proposed modification this may not be needed anymore as the transformation always would return root? This could be:
```
for routine in psyir.walk(Routine):
    root = invoke_trans.apply(routine)
```

Reply:
Yes, it's simpler now.